src/main/java/com/mia/BuhaiCommunications/controller/rest/UserRestController.java

```diff
@@ -1,5 +1,52 @@
 package com.mia.BuhaiCommunications.controller.rest;
 
+import com.mia.BuhaiCommunications.model.User;
+import com.mia.BuhaiCommunications.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api")
 public class UserRestController {
-    //TODO: create functionality
+    @Autowired
+    private UserRepository userRepository;
+
+    @GetMapping("/users")
+    public ResponseEntity<List<User>> getAllUsers() {
+        return ResponseEntity.ok(userRepository.findAll());
+    }
+
+    @GetMapping("/user/{id}")
+    public ResponseEntity<User> getUser(@PathVariable Integer id) {
+        return ResponseEntity.of(userRepository.findById(id));
+    }
+
+    @PostMapping("/message/create")
+    public ResponseEntity<User> createUser(@RequestBody User user) {
+        userRepository.save(user);
+        return ResponseEntity.ok(user);
+    }
```

Comment:
Shouldn't the PostMapping here be "/user/create" ?
"/message/create" seems misleading to me.